astor/code_gen.py

```diff
@@ -46,9 +47,15 @@ def to_source(node, indent_with=' ' * 4, add_line_information=False,
     of the nodes are added to the output.  This can be used to spot wrong line
     number information of statement nodes.
 
+    `source_generator_class` defaults to `None`, and specifies the
+    class that will be instantiated and used to generate the source code.
+    When `None`, it will use `SourceGenerator`.
+
     """
-    generator = SourceGenerator(indent_with, add_line_information,
-                                pretty_string)
+    if source_generator_class is None:
+        source_generator_class = SourceGenerator
+    generator = source_generator_class(
```

Comment:
Could you also raise a `TypeError` if `source_generator_class` is not a subclass of `SourceGenerator`?